src/config.rs

```diff
@@ -0,0 +1,290 @@
+use anyhow::{Context, Result};
+use hcl::Value;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::path::Path;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HealthChecks {
+    pub listener_enabled: Option<bool>,
+    pub bind_port: Option<u16>,
+    pub liveness_path: Option<String>,
+    pub readiness_path: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct JwtSvid {
+    pub jwt_audience: String,
+    pub jwt_extra_audiences: Option<Vec<String>>,
+    pub jwt_svid_file_name: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Config {
+    pub agent_address: Option<String>,
+    pub cmd: Option<String>,
+    pub cmd_args: Option<String>,
+    pub pid_file_name: Option<String>,
+    pub cert_dir: Option<String>,
+    pub daemon_mode: Option<bool>,
+    pub add_intermediates_to_bundle: Option<bool>,
+    pub renew_signal: Option<String>,
+    pub svid_file_name: Option<String>,
+    pub svid_key_file_name: Option<String>,
+    pub svid_bundle_file_name: Option<String>,
+    pub jwt_svids: Option<Vec<JwtSvid>>,
+    pub jwt_bundle_file_name: Option<String>,
+    pub include_federated_domains: Option<bool>,
+    pub cert_file_mode: Option<String>,
+    pub key_file_mode: Option<String>,
+    pub jwt_bundle_file_mode: Option<String>,
+    pub jwt_svid_file_mode: Option<String>,
+    pub hint: Option<String>,
+    pub omit_expired: Option<bool>,
+    pub health_checks: Option<HealthChecks>,
+}
+
+pub fn parse_hcl_config(path: &Path) -> Result<Config> {
+    let content = fs::read_to_string(path)
+        .with_context(|| format!("Failed to read config file: {}", path.display()))?;
+
+    let value: Value = hcl::from_str(&content)
+        .with_context(|| format!("Failed to parse HCL config file: {}", path.display()))?;
+
+    parse_hcl_value_to_config(&value)
+}
+
+fn parse_hcl_value_to_config(value: &Value) -> Result<Config> {
```

Comment:
@cursoragent I prefer to use `hcl:Value` here instead of `Value`. Include the crate name makes the code more readable.